blmath/optimization/objectives/normals.py

```diff
@@ -0,0 +1,242 @@
+# pylint: disable=arguments-differ, invalid-unary-operand-type
+import numpy as np
+import chumpy as ch
+import scipy.sparse as sp
+from blmath.numerics.matlab import row, col
+
+try:
+    from opendr.geometry import TriNormals, VertNormals, TriNormalsScaled, NormalizedNx3  # This lets us import bodylabs.geometry.normals.NormalizedNx3 etc. whether or not opendr is available. pylint: disable=unused-import
```

Comment:
I'm inclined to leave as such for now, since tons of code is moving in this ticket already.